tools/GdkTestRunner/Modules/UnityModule.cs

```diff
@@ -0,0 +1,166 @@
+        private readonly string logfilePath;
+        private readonly string testResultsPath;
+
+        private readonly Logger logger;
+
+        public UnityModule(GdkTestRunnerOptions options, JToken jsonContext) : base(options, jsonContext)
+        {
+            var parsedJson = jsonContext.ToObject<UnityModuleDefinition>();
+
+            Name = parsedJson.Name;
+            unityProjectPath = Path.Combine(Path.GetDirectoryName(options.ConfigurationFilePath),
+                parsedJson.UnityDefinition.UnityProjectPath);
+            unityTestPlatform = parsedJson.UnityDefinition.TestPlatform;
+
+            if (!ValidateTestPlatform())
+            {
+                throw new ArgumentException($"Unknown test platform: {unityTestPlatform}");
+            }
+
+            var formattedName = TitleCaseToKebabCase(parsedJson.Name);
+
+            logfilePath = GetLogFilePath($"{formattedName}-{unityTestPlatform}.log");
+            testResultsPath = GetLogFilePath($"{formattedName}-{unityTestPlatform}-results.xml");
+
+            logger = LogManager.GetCurrentClassLogger();
+        }
+
+        protected override void BeforeRun()
+        {
+            CleanLibraryAndTempFolders();
+        }
+
+        protected override bool InternalRun()
+        {
+            var currentDirectory = Environment.CurrentDirectory;
+            Environment.CurrentDirectory = unityProjectPath;
+
+            var unityPath = GetUnityExePath(Paths.TryGetUnityPath());
+            var arguments = new[]
+            {
+                "-batchmode",
+                "-projectPath", $"\"{unityProjectPath}\"",
+                "-runTests",
+                "-testPlatform", $"{unityTestPlatform}",
+                "-logfile", $"\"{logfilePath}\"",
+                "-testResults", $"\"{testResultsPath}\""
+            };
+
+            var args = string.Join(' ', arguments);
+
+            logger.Debug($"Running: {unityPath} {args}");
+
+            try
+            {
+                if (!RunProcess(unityPath, args))
+                {
+                    logger.Error($"{Name} exited with a non-zero exit code. " +
+                        $"Check {logfilePath} and {testResultsPath} for more info.");
+                    return false;
+                }
+            }
+            catch (Exception e)
+            {
+                logger.Error($"Exception thrown while testing: {e.Message}\n{e.InnerException.Message}");
+                return false;
+            }
+            finally
+            {
+                Environment.CurrentDirectory = currentDirectory;
+            }
+
+            return true;
+        }
+
+        protected override void AfterRun()
+        {
+            CleanLibraryAndTempFolders();
+        }
+
+        private void CleanLibraryAndTempFolders()
+        {
+            var libraryFolder = Path.Combine(unityProjectPath, "Library");
+            var tempFolder = Path.Combine(unityProjectPath, "Temp");
```

Comment:
should we also delete the obj folder?

Reply:
This was pulled from what we currently do in `test.sh`. Not sure what the implications of deleteing one and not the other is

Reply:
would delete obj as well just to make sure. they are all temporary folders that might appear and should be cleaned up